### PR TITLE
FEATURE: DC-2529 Add documentation for JWT authentication 

### PR DESCRIPTION
**Pull request by xMort**
JIRA: DP-2529

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15158/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16704/ : SUCCESS in 6m 52s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15162/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16708/ : SUCCESS in 6m 49s

File: docs/content/en/latest/learn/integrate_and_authenticate/CN_and_cloud_authentication.md [hunk 1]
```diff
@@ -48,6 +48,55 @@ called whenever the **NotAuthenticated exception** is raised by the backend.
     }, 500);
 ```
 
+## JWT authentication
+
+This type of authentication uses JWT (JSON Web Token) as bearer of authentication. Unlike API Token, JWT has expiration period.
```
Comment (Mara3l) [inline docs/content/en/latest/learn/integrate_and_authenticate/CN_and_cloud_authentication.md:53]:
```suggestion
This type of authentication uses JWT (JSON Web Token) as bearer of authentication. Unlike an API Token, JWT has an expiration period.
```

File: docs/content/en/latest/learn/integrate_and_authenticate/CN_and_cloud_authentication.md [hunk 2]
```diff
@@ -48,6 +48,55 @@ called whenever the **NotAuthenticated exception** is raised by the backend.
...
+
+You can read more about it at the [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519) page or [here](https://jwt.io/introduction).
+
+You can read [help page](https://www.gooddata.com/developers/cloud-native/doc/cloud/manage-organization/jwt-access-token/) that will guide you
+through the process of generating JWK (JSON Web Key) and registering it in your organization. It covers generating of JWT as well.
```
Comment (Mara3l) [inline docs/content/en/latest/learn/integrate_and_authenticate/CN_and_cloud_authentication.md:58]:
```suggestion
through the process of generating JWK (JSON Web Key) and registering it in your organization. It covers the generation of JWT as well. 
```

File: docs/content/en/latest/learn/integrate_and_authenticate/CN_and_cloud_authentication.md [hunk 3]
```diff
@@ -48,6 +48,55 @@ called whenever the **NotAuthenticated exception** is raised by the backend.
...
+You can read more about it at the [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519) page or [here](https://jwt.io/introduction).
+
+You can read [help page](https://www.gooddata.com/developers/cloud-native/doc/cloud/manage-organization/jwt-access-token/) that will guide you
+through the process of generating JWK (JSON Web Key) and registering it in your organization. It covers generating of JWT as well. 
+The guide uses GoodData Python SDK. Alternatively, you can follow [JavaScript guide](https://www.gooddata.com/developers/cloud-native/doc/cloud/manage-organization/jwt-access-token/jwt-javascript-example/) 
+to achieve the same via a Node.JS script.
+
+It's meant to be used in GoodData Cloud and GoodData.CN UI applications.
+ 
+The example how to set up the `AnalyticalBackend` instance with `TigerJwtAuthProvider` authentication provider:
```
Comment (Mara3l) [inline docs/content/en/latest/learn/integrate_and_authenticate/CN_and_cloud_authentication.md:64]:
```suggestion
The example of how to set up the `AnalyticalBackend` instance with `TigerJwtAuthProvider` authentication provider:
```

**Review by Mara3l — CHANGES_REQUESTED** [review]
Looks good! I just have some minor suggestions. :)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15235/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16766/ : SUCCESS in 6m 43s

**Review by Mara3l — APPROVED** [review]

**Review by ivan-nejezchleb — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3987

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3093/ : SUCCESS in 11m 51s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16779/ : SUCCESS in 6m 48s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/668/ : SUCCESS in 11m 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4251/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/973/ : FAILURE in 1m 20s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/605/ : SUCCESS in 7m 46s

**Pull request merged**